(maint) Update to new signing key

melissa opened the pull request.

melissa committed:
(maint) Update to new signing key

This commit adds the new signing key as well as updates the old signing
key to use the long keyid for added security

melissa commented:
cc/ @highb @ericwilliamson @haus you three have made a few recent commits to this repo. I want to check to ensure this isn't dead code and it worth updating. Thanks!

haus commented:
@melissa deprecated, but still shipped in PE 3.8 afaik

haus commented:
@melissa looks like the yum repo definition needs updating here too

melissa commented:
Oops, that's for pointing that out! I'll get that updated now

melissa commented:
Updated. PE 3.8 will continue to use the new key, but if we're shipping this module and expect it to work with the open source repos, we'll need this change

haus commented on lib/puppet/cloudpack/scripts/master_source_dev.erb:
```diff
@@ -13,7 +13,7 @@ if [ -f /etc/redhat-release ]; then
 baseurl=http://yum.puppetlabs.com/el/\$releasever/dependencies/\$basearch
 enabled=1
-gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
+gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppet
```
I think you'll want to add the new key and not replace it here (and elsewhere), as older versions of puppet will still be signed with the old key.

```gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
                http://yum.puppetlabs.com/RPM-GPG-KEY-puppet```

^^ is the right syntax according to https://projects.puppetlabs.com/issues/8867.

haus commented on lib/puppet/cloudpack/scripts/puppet-community.erb:
```diff
@@ -43,7 +43,7 @@ function rpm_install() {
 function apt_install() {
   # Download and install the puppetlabs apt public
-  apt-key adv --recv-key --keyserver pool.sks-keyservers.net 4BD6EC30
+  apt-key adv --recv-key --keyserver pool.sks-keyservers.net 7F438280EF8D349F
```
This keyserver seems to be toast afaict. Might be worth replacing with `pgp.mit.edu`?

melissa commented:
Updated! I thought I had pushed the update out yesterday, but apparently I never did...

ericwilliamson commented:
@melissa Just a reminder that this module is not shipped as part of the PE modules, but as part of the cloud provisioner package..I think

melissa commented:
This code will still fail once we switch over to the new key, so it'll need these changes. The open source repos will all have the new key

shrug commented on lib/puppet/cloudpack/scripts/master_source_dev.erb:
```diff
@@ -14,6 +14,7 @@ name=Puppet Labs PS (Unofficial)
 baseurl=http://yum.puppetlabs.com/el/\$releasever/dependencies/\$basearch
 enabled=1
 gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
+       http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
```
Looks like this duplicates the legacy key?

melissa replied:
womp womp. Fixed!

shrug commented:
👍  from me. Those test failures don't look related...

haus commented on lib/puppet/cloudpack/scripts/puppet-community.erb:
```diff
@@ -44,6 +47,7 @@ function rpm_install() {
 function apt_install() {
   # Download and install the puppetlabs apt public
   apt-key adv --recv-key --keyserver pool.sks-keyservers.net 4BD6EC30
```
want to replace this with the long key id as long as you're in here?

melissa replied:
Done

haus commented:
gem versions are fucking travis up

haus commented:
"Gem::InstallError: mime-types-data requires Ruby version >= 2.0.
An error occurred while installing mime-types-data (3.2016.0521), and Bundler" < from 1.9.3 cell

melissa commented:
Travis failures are unrelated to these changes. Merge?

haus commented:
I'm a fan of fixing broken shit when it gets seen, but how about at least opening a ticket for the travis failures?

melissa committed:
(maint) Pin to google-api-client version

Prior to this commit, we had been pulling in the latest version of
google-api-client. This is generally fine, except that gem has a runtime
dependency `mime-types >= 1.6`, which has a runtime dependency
`mime-types-data ~> 3.2015` which cannot be installed because it
requires Ruby >= 2.0. We need to be able to run these tests on Ruby
1.9.3, so we need to pin to an older version of google-api-client that
doesn't have a dependency chain that requires Ruby 2.0.

melissa commented on Gemfile:
```diff
@@ -15,7 +15,7 @@ end
 gem 'fog', :require => false
 gem 'guid', :require => false
-gem 'google-api-client', :require => false
+gem 'google-api-client', "~> 0.8.7", :require => false
```
@haus fine, I fixed it.

melissa replied:
Well, I fixed the gem version failure, but not the spec test failure.

On ruby 1.9.3:
```
/home/melissa/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/fog-profitbricks-2.0.1/lib/fog/profitbricks.rb:4:in `<top (required)>': undefined local variable or method `__dir__' for main:Object (NameError)
	from /home/melissa/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/fog-1.38.0/lib/fog.rb:51:in `require'
	from /home/melissa/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/fog-1.38.0/lib/fog.rb:51:in `<top (required)>'
	from /home/melissa/puppetlabs-cloud_provisioner/lib/puppet/cloudpack.rb:4:in `require'
...
```

On ruby 2.0.0:
```
/home/melissa/puppetlabs-cloud_provisioner/lib/puppet/cloudpack.rb:5:in `require': cannot load such file -- net/ssh (LoadError)
	from /home/melissa/puppetlabs-cloud_provisioner/lib/puppet/cloudpack.rb:5:in `<top (required)>'
	from /home/melissa/puppetlabs-cloud_provisioner/spec/spec_helper.rb:15:in `require'
	from /home/melissa/puppetlabs-cloud_provisioner/spec/spec_helper.rb:15:in `<top (required)>'
...
```

Do you want me to try to address those failures as well?

haus commented:
@melissa the 1.9.3 failure looks like a problem in fog-profitbricks, which is being fixed (but looks unreleased) in https://github.com/fog/fog-profitbricks/commit/f5c25e38b43331f7f0771e24517681555f9583fc

haus commented:
@melissa oh my god that is a sad road full of redness. I'm just gonna merge this. Getting tests to pass is a nightmare.

melissa commented:
Thank you Haus. What are the next steps for this? Is there CI or should I make sure it's promoted into something?

haus commented:
@melissa i have no idea. but it needs to be built for PE 3.8. probably via enterprise-dist rake tasks.
